src/solvers/thermal_solver.hpp

```diff
@@ -22,38 +22,41 @@ class DynamicConductionOperator;
 class ThermalSolver : public BaseSolver
 {
 protected:
+
+  FiniteElementState & temperature;
+
   /// Mass bilinear form object
-  mfem::ParBilinearForm *m_M_form;
+  std::shared_ptr < mfem::ParBilinearForm > m_M_form;
 
   /// Stiffness bilinear form object
-  mfem::ParBilinearForm *m_K_form;
+  std::shared_ptr < mfem::ParBilinearForm > m_K_form;
 
   /// Assembled mass matrix
-  mfem::HypreParMatrix *m_M_mat;
+  std::shared_ptr < mfem::HypreParMatrix > m_M_mat;
 
   /// Eliminated mass matrix
-  mfem::HypreParMatrix *m_M_e_mat;
+  std::shared_ptr < mfem::HypreParMatrix > m_M_e_mat;
 
   /// Assembled stiffness matrix
-  mfem::HypreParMatrix *m_K_mat;
+  std::shared_ptr < mfem::HypreParMatrix > m_K_mat;
 
   /// Eliminated stiffness matrix
-  mfem::HypreParMatrix *m_K_e_mat;
+  std::shared_ptr < mfem::HypreParMatrix > m_K_e_mat;
 
   /// Thermal load linear form
-  mfem::ParLinearForm *m_l_form;
+  std::shared_ptr < mfem::ParLinearForm > m_l_form;
 
   /// Assembled BC load vector
-  mfem::HypreParVector *m_bc_rhs;
+  std::shared_ptr < mfem::HypreParVector > m_bc_rhs;
 
   /// Assembled RHS vector
-  mfem::HypreParVector *m_rhs;
+  std::shared_ptr < mfem::HypreParVector > m_rhs;
 
   /// Linear solver for the K operator
-  mfem::CGSolver *m_K_solver;
+  std::shared_ptr < mfem::CGSolver > m_K_solver;
 
   /// Preconditioner for the K operator
-  mfem::HypreSmoother *m_K_prec;
+  std::shared_ptr < mfem::HypreSmoother > m_K_prec;
 
   /// Conduction coefficient
   mfem::Coefficient *m_kappa;
```

Comment:
Is there a reason m_kappa and m_source didn't get turned into shared? Should the reason be in a comment?

Reply:
Good question, I was thinking of adopting a convention: std::shared_ptr / std::unique_ptr for pointers where we own the data, and raw pointers / references for views (kappa and source aren't given in the constructor, so we can't initialize reference variables there).

Comments to explain that are a good idea though!